openmc/mgxs/mdgxs.py

```diff
@@ -0,0 +1,1448 @@
+            xs = xs_tally.get_values(filters=filters,
+                                     filter_bins=filter_bins, value=value)
+        else:
+            xs = self.xs_tally.get_values(filters=filters,
+                                          filter_bins=filter_bins,
+                                          nuclides=query_nuclides, value=value)
+
+        # Divide by atom number densities for microscopic cross sections
+        if xs_type == 'micro':
+            if self.by_nuclide:
+                densities = self.get_nuclide_densities(nuclides)
+            else:
+                densities = self.get_nuclide_densities('sum')
+            if value == 'mean' or value == 'std_dev':
+                xs /= densities[np.newaxis, :, np.newaxis]
+
+        # Reverse data if user requested increasing energy groups since
+        # tally data is stored in order of increasing energies
+        if order_groups == 'increasing':
+            if groups == 'all':
+                num_groups = self.num_groups
+            else:
+                num_groups = len(groups)
+
+            # Reshape tally data array with separate axes for domain and energy
+            num_subdomains = int(xs.shape[0] / num_groups)
+            new_shape = (num_subdomains, num_groups) + xs.shape[1:]
+            xs = np.reshape(xs, new_shape)
+
+            # Reverse energies to align with increasing energy groups
+            xs = xs[:, ::-1, :]
+
+        # Eliminate trivial dimensions
+        xs = np.squeeze(xs)
+        xs = np.atleast_1d(xs)
+        return xs
+
+    def get_slice(self, nuclides=[], groups=[], delayed_groups=[]):
+        """Build a sliced MDGXS for the specified nuclides, energy groups,
+        and delayed groups.
+
+        This method constructs a new MDGXS to encapsulate a subset of the data
+        represented by this MDGXS. The subset of data to include in the tally
+        slice is determined by the nuclides, energy groups, delayed groups
+        specified in the input parameters.
+
+        Parameters
+        ----------
+        nuclides : list of str
+            A list of nuclide name strings
+            (e.g., ['U-235', 'U-238']; default is [])
+        groups : list of int
+            A list of energy group indices starting at 1 for the high energies
+            (e.g., [1, 2, 3]; default is [])
+        delayed_groups : list of int
+            A list of delayed group indices
+            (e.g., [1, 2, 3]; default is [])
+
+        Returns
+        -------
+        openmc.mgxs.MDGXS
+            A new MDGXS object which encapsulates the subset of data requested
+            for the nuclide(s) and/or energy group(s) and/or delayed group(s)
+            requested in the parameters.
+
+        """
+
+        cv.check_iterable_type('nuclides', nuclides, basestring)
+        cv.check_iterable_type('energy_groups', groups, Integral)
+        cv.check_type('delayed groups', delayed_groups, list, int)
```

Comment:
ditto - perhaps `delayed_groups` can be more generally be any `Iterable` but it's nbd